engine/apps/api/views/user.py

```diff
@@ -253,9 +268,24 @@ def get_queryset(self):
 
     def list(self, request, *args, **kwargs) -> Response:
         queryset = self.filter_queryset(self.get_queryset())
+
+        is_currently_oncall_query_param = request.query_params.get("is_currently_oncall", "").lower()
+
+        def _get_oncall_user_ids():
+            return {user.pk for _, users in self.schedules_with_oncall_users.items() for user in users}
+
+        if is_currently_oncall_query_param == "true":
+            # client explicitly wants to filter out users that are on-call
+            queryset = queryset.filter(pk__in=_get_oncall_user_ids())
+        elif is_currently_oncall_query_param == "false":
+            # user explicitly wants to filter out on-call users
+            queryset = queryset.exclude(pk__in=_get_oncall_user_ids())
```

Comment:
I'm not 100% sure whether behaviour like this is better suited to live in the `FilterSet` class? I put it here because I was unsure how, within the `UserFilter` class, to get access to `self.schedules_with_oncall_users` (or at a minimum, `self.request.user`)